### PR TITLE
try build on linux

### PR DESCRIPTION
**Pull request by bgruening**
I'm not sure why mono still have this limitation of the limited path length. Maybe there is a compile switch for mono as well?

However, in this PR we should try to limit the path length for the time being?

ping @m-rossi

### CODE REVIEW

**Comment by bgruening** [comment]
@conda-forge-admin, please re-render

**Comment by bgruening** [comment]
Mh, it seems to build but crashes badly during testing, dang! :(

**Comment by m-rossi** [comment]
I think I understand some parts of that issue. I was looking at the travis scripts and found the following lines: https://github.com/pythonnet/pythonnet/blob/af419b10e191f6c7673524b4a2dc5e8dab7643dc/.travis.yml#L93-L96 and added some flavor of it to the `build.sh`:
```bash
PY_LIBDIR=$($PYTHON -c 'import sysconfig; print(sysconfig.get_config_var("LIBDIR"))')
export LD_LIBRARY_PATH=/opt/conda/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib/:$PY_LIBDIR:$LD_LIBRARY_PATH
```
then the building produces an `*.so` for the `clr` inside the package. However the import still fails. Only if I execute the above lines before the import it works. So importing the package relies on being the compiler present at runtime? Does this also apply to other Linux distributions or only for the CentOS based CI-image?

**Comment by bgruening** [comment]
@m-rossi that should not be needed imho. If you import `{{ compiler('c') }}` during build, libgcc will be added automatically to your run-time environment. However, libgcc is not gcc. I added the full compiler stack in my last commit to the run time environment. Lets see, but this should not be needed.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,31 +10,35 @@ source:
...
+    - mono                 # [not win]
     - python
 
 test:
   imports:
-    - clr
+    - clr  # [not linux]
+  commands:
+    - PY_LIBDIR=$($PYTHON -c 'import sysconfig; print(sysconfig.get_config_var("LIBDIR"))')  # [linux]
+    - export LD_LIBRARY_PATH=$PY_LIBDIR:$PREFIX/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib:$LD_LIBRARY_PATH  # [linux]
```
Comment (bgruening) [inline recipe/meta.yaml]: I don't think we should merge this PR even if its works. This is super wired. At worst we should move this to an env activation hook.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,31 +10,35 @@ source:
...
     - glib  # [not win]
     - mono  # [not win]
     - pip
     - pycparser
     - python
   run:
-    - gettext  # [not win]
-    - libiconv  # [not win]
-    - mono  # [not win]
+    - {{ compiler('c') }}  # [linux]
```
Comment (bgruening) [inline recipe/meta.yaml]: I guess then the compiler is not needed here.

File: recipe/meta.yaml (hunk 1 above)
Comment (m-rossi) [reply recipe/meta.yaml]: Me too, I just wanted to try if we get it working.

Comment (bgruening) [reply recipe/meta.yaml]: Uha, there are a few more in conda-forge that already had this problem:

https://github.com/conda-forge/r-pca3d-feedstock/commit/7e9fb5d0c603eb150fcd2d56a8f36952c1ca313e

Comment (bgruening) [reply recipe/meta.yaml]: Interestingly this package works normally so maybe this is really a conda-forge build/test thing.

Can you mirror the commit above and test this?

Comment (m-rossi) [reply recipe/meta.yaml]: What should I mirror?
* The test command with setting the environment variable? It’s quite similar to mine. 
* The CDT packages? Those shouldn‘t help us at runtime, or?

We could remove the compiler from the runtime requirements and use the compiler dir from the build environment

Comment (bgruening) [reply recipe/meta.yaml]: The test section and the build.sh script.

Comment (m-rossi) [reply recipe/meta.yaml]: Seems the variable is not needed in the build-script. I removed the runtime dependency of the compiler and simplified the testing commands.

**Review by bgruening — COMMENTED** [review]
Cool, thanks a bunch!
Let's try this build :)

**Pull request merged**